USF-1052: Cart feature matrix

jeff-matthews opened the pull request:
This pull request (PR) adds a table showing which Adobe Commerce features that the cart dropin supports.

>[!NOTE]
>The `OptionsTable` component doesn't currently support image links. It would be nice to have icons/.svg files for each status similar to [this](https://developer.adobe.com/commerce/pwa-studio/integrations/adobe-commerce/features/).

I know that we're attempting to finalize the dropin doc template (#41), but since we'll have to refactor the cart docs after that anyway, it seems prudent to get this info out on production. It's unclear how long it will take to get the template finalized.

jeff-matthews committed:
docs: added initial draft of cart feature matrix

bdenham reviewed (approved):
Nice! Looks great. Thanks!

bdenham commented on src/content/docs/dropins/cart/index.mdx:
```diff
@@ -7,12 +7,37 @@ sidebar:
-<Aside type="wip" title="In Development">
-  The Cart dropin is in active development. Stay tuned for updates.
-</Aside>
+The following table provides an overview of the Adobe Commerce features that the Cart dropin supports:
```
```suggestion
## Commerce features supported

The following table provides an overview of the Adobe Commerce features that the Cart dropin supports:
```

bdenham reviewed (approved):
Suggested same heading for this table as in Checkout feature matrix.

jeff-matthews committed:
docs: apply suggestions from code review

Co-authored-by: Bruce Denham <bdenham@adobe.com>

jeff-matthews committed:
docs: apply suggestions from code review

jeff-matthews committed:
Merge branch 'dropin-feature-matrices' into USF-1052-cart-feature-matrix

jeff-matthews committed:
fix: converted to markdown tables w/ badges

jeff-matthews commented:
I converted to the default Starlight table component and added badges. It required some minor style adjustments, which I [made](https://github.com/commerce-docs/microsite-commerce-storefront/commit/f121f7f02a121378a2ffe8f7397e974f505207ab) in a separate integration branch (now the base branch for this PR):

![cart](https://github.com/user-attachments/assets/d0def125-ec3e-4d03-9a55-c32c8791e97e)

bdenham reviewed (approved):
Excellent use of the Badge component.
